design/ceph-csi-operator.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Integrate cephcsi operator with Rook
+target-version: release-1.15
+---
+
+# Integrate cephcsi operator with Rook
+
+## Overview
+
+To integrate the new ceph-csi operator with Rook and remove the existing csi deployment related code, RBAC's and dead code in later releases. Also before moving forward please read the ceph-csi-operator design doc [here](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md).
+
+## Rook Deploying Ceph-CSI-Operator
+
+There are two scenarios rook needs to manage while deploying of the new csi-operator.
+
+1. Deployment in new cluster i.e [Fresh deployment](#fresh-deployment)
+    1. Manifest Deployment
+    2. Helm Deployment
+2. Deployment in upgrade cluster i.e [Upgrade clusters](#upgraded-clusters) cluster.
+    1. Manifest Upgrade
+    2. Helm Upgrade
+
+We need to have some bash script which will be called by some make command to automatically update the RBAC, CRDs files in rook repo from the ceph-csi-operator repo.
+
+## Fresh deployment
+
+Rook will define the new csi-operator defaults settings in [CRDs](#new-ceph-csi-operator-crds) in operator.yaml or operator chart `rook-ceph` `value.yaml`, and can be customizable by users. Rook will define the CSI operator and related CRs such as it is deployed by default and automatically with Rook.
+
+- **Manifests**:
```

Comment:
This is the ideal for stable mode. For experimental, let's have a separate manifest (or multiple if needed).

Reply:
Does this mean we want to keep, crds, rbac and examples all in different files?
csi-operator crds/rbac will not be included in rook crds.yaml and common.yaml

Reply:
For experimental mode, it's fine to have separate files. We can still have a single command for them to install all the manifests in a subdirectory, such as:
```
kubectl create -f deploy/examples/csi-operator/
```

Reply:
But for stable mode, I would like to see these merged with our existing manifests so no additional manifests if possible.

Reply:
Yes, in experimental it will individual files and in future release it will be merged with rook manifests